src/client/datascience/jupyter/kernels/cellExecutionQueue.ts

```diff
@@ -132,4 +171,28 @@ export class CellExecutionQueue {
             }
         }
     }
+
+    private getCellExecutions(executionList: Execution[]): CellExecution[] {
```

Comment:
Not a perf difference I believe, but this could look cleaner just using .filter

Reply:
I agree, personally we don't need these two methods, i'd just move the filtering into the place where we call this method, else we just iterate throught the same set of list twice.